crates/diesel_models/src/payment_method.rs

```diff
@@ -105,6 +105,175 @@ impl PaymentMethod {
         &self.payment_method_id
     }
 
+    pub fn deserialize<'de, D>(deserializer: D) -> Result<Option<PaymentMethod>, D::Error>
```

Comment:
We don't need to implement a deserializer for this.

We need to implement a fn which takes in serde_json::Value, and returns back a `CommonMandateReference` after performing the required steps

Reply:
`pub fn get_common_mandate_reference(
        connector_mandate_details: Option<serde_json::Value>,
    ) -> Result<CommonMandateReference, ParsingError> {
        if let Some(value) = connector_mandate_details {
            match serde_json::from_value::<CommonMandateReference>(value.clone()) {
                Ok(common_mandate_reference) => Ok(common_mandate_reference),
                Err(_) => match serde_json::from_value::<PaymentsMandateReference>(value.clone()) {
                    Ok(payment_mandate_reference) => Ok(CommonMandateReference {
                        payments: Some(payment_mandate_reference),
                        payouts: None,
                    }),
                    Err(_) => Err(ParsingError::StructParseFailure(
                        "Faild to deserialize PaymentMethod",
                    ))?,
                },
            }
        } else {
            Err(ParsingError::StructParseFailure(
                "Faild to deserialize PaymentMethod",
            ))?
        }
    }`
    
    added

Reply:
`pub fn get_common_mandate_reference(
        connector_mandate_details: Option<serde_json::Value>,
    ) -> Result<CommonMandateReference, ParsingError>` added